### DISCUSSION TITLE
How reliable is a WhatsApp Gateway?

### DISCUSSION THREAD

**Discussion by mikedolx** [42wim/matterbridge#1497] [Help / Q&A] [answered]
Hi,

I'm currently evaluating the abilities of matterbridge - especially the WhatsApp gateway. I already set up a bridge and made my first tests. But before I go into the technical details, I'd like to ask some general questions.
Here my questions. Some might be stupid or obvious, but I'd like to clear that out. Hope you can help me.

-  Does anybody has a long term experience with the WA protocol? How long (weeks, months, years) did you ran the bridge without interruption? (restarting the bridge, recreating the session.gob file)
- Do I need to keep the main WA-Client alive (the one I used to scan the QR-code)  
- How often can I reuse the session.gob file for restarting the bridge?

Thanks

**Comment by ciroiriarte** [comment]
Right now I'm having issues with the session restore. It fails monthly, you re-auth and then it continues but for "unattended operation" it doesn't work.

If you can easily reath monthly, could we "good enough"

ref: https://github.com/Rhymen/go-whatsapp/issues/268

**Reply by mikedolx** [reply]
I had it now running for over two months, but with interruptions. I think i had to restart the bridge at lease once every two weeks. The bridge was between Telegram and WhatsApp.
In think, that the reliability heavily relies on the WhatsApp-Phone. I saw, that once after WhatsApp received an automatic update via AppStore the connection was interrupted.

But now they introduce a new devices feature, which is currently beta. Each of the clients can be online up to two weeks, without the actual phone to be switched on.

For now, i'm evaluating Matrix with the mautrix-whatsapp bridge. It seems, that convenience whise the matrix concept is much better. Once setup, you can control the bridge via a bot-chat. You have commands like login, logout or ping. That is from my point of view much more conveniant, than ssh'ing into a server in case the bridge goes down. 
I'll see how long the mautrix-whatsapp bridge will go.

**Comment by UncleAndy123** [comment]
Hi @mikedolx!
Just stumbled onto this post, and wondered if you would be interested in providing us another update. How is the reliability after several years?

**Reply by mikedolx** [reply, marked as answer]
Hi @UncleAndy123 ,

very quickly after making my first trials with the matterbridge, I switched to a matrix server with a mateix-to-WhatsApp bridge. 
The reason for that was the handling of contacts an new groups. Each time i wanted to initiate a new chat I hat to set this up via terminal on my pc/laptop.
Using the matrix-WhatsApp-bridge I could do that, by chatting with a dedicated bot.
Meanwhile I gave up hosting a matrix-server as it was too much doing that just for me :-).

I could do another attempt and dig into the matterbridge and check if managing contacts and groups has improved over the time.

Hope that answer helps.

Michael;
